client/src/helpers/APICalls/requests.ts

```diff
@@ -0,0 +1,51 @@
+import { FetchOptions } from '../../interface/FetchOptions';
+import { Request, RequestsList } from '../../interface/Request';
+import { User } from '../../interface/User';
+
+export async function getRequestList(): Promise<RequestsList> {
+  const fetchOptions: FetchOptions = {
+    method: 'GET',
+    credentials: 'include',
+  };
+
+  return await fetch('/request/', fetchOptions).then((res) =>
+    res.json().catch(() => ({ error: { message: 'Unable to connect to server. Please try again' } })),
```

Comment:
You are not chaining `catch(() =>` the correct place. You need to chain it like this: `.then(...).catch(...)`. As long as each `then` executes a `return` statement, all Promise rejections will be caught in one place. But here you're only `catch`ing the rejection from `json()` and not that from `fetch()`